qforce/fragment.py

```diff
@@ -357,26 +369,31 @@ def check_for_qm_data(self, job, config, mol, qm):
 
         else:
             self.check_new_scan_data(job, mol, config, qm)
-            self.write_have_or_missing(job)
+            self.write_have_or_missing(job, config)
             nx.write_gpickle(self.graph, f"{self.dir}/identifier_{self.hash_idx}")
             self.write_xyz()
             with open(f"{self.dir}/qm_method_{self.hash_idx}", 'w') as file:
                 json.dump(self.graph.graph['qm_method'], file, sort_keys=True, indent=4)
 
             if not self.has_data:
```

Comment:
if not (self.has_data or (config.batch_run and self.has_inp)):

Reply:
Sorry, I don't quite understand this part.

Reply:
Sorry that was a bit ambiguous. I meant replace:

```
            if not self.has_data:
                self.make_qm_input(job, qm)
                if not (config.batch_run and self.has_inp):
                    self.make_qm_input(job, qm)
```
with:

```
            if not (self.has_data or (config.batch_run and self.has_inp)):
                self.make_qm_input(job, qm)
```